ngraph/core/src/type/element_type.cpp

```diff
@@ -12,45 +12,47 @@
 #include "ngraph/type/element_type_traits.hpp"
 
 using namespace ngraph;
-using namespace std;
 
 constexpr DiscreteTypeInfo AttributeAdapter<element::Type>::type_info;
-
-class TypeInfo
+namespace
 {
-public:
-    TypeInfo(size_t bitwidth,
-             bool is_real,
-             bool is_signed,
-             bool is_quantized,
-             const std::string& cname,
-             const std::string& type_name)
-        : m_bitwidth{bitwidth}
-        , m_is_real{is_real}
-        , m_is_signed{is_signed}
-        , m_is_quantized{is_quantized}
-        , m_cname{cname}
-        , m_type_name{type_name}
+    class TypeInfo
     {
-    }
-    size_t m_bitwidth;
-    bool m_is_real;
-    bool m_is_signed;
-    bool m_is_quantized;
-    std::string m_cname;
-    std::string m_type_name;
-};
+    public:
+        TypeInfo(size_t bitwidth,
+                 bool is_real,
+                 bool is_signed,
+                 bool is_quantized,
+                 const std::string& cname,
+                 const std::string& type_name)
+            : m_bitwidth{bitwidth}
+            , m_is_real{is_real}
+            , m_is_signed{is_signed}
+            , m_is_quantized{is_quantized}
+            , m_cname{cname}
+            , m_type_name{type_name}
+        {
+        }
+        size_t m_bitwidth;
+        bool m_is_real;
+        bool m_is_signed;
+        bool m_is_quantized;
+        std::string m_cname;
+        std::string m_type_name;
+    };
 
-struct element_type_hash
-{
-    size_t operator()(element::Type_t t) const { return static_cast<size_t>(t); }
-};
+    struct ElementTypes
+    {
+        struct element_type_hash
+        {
+            size_t operator()(element::Type_t t) const { return static_cast<size_t>(t); }
+        };
 
-typedef unordered_map<element::Type_t, const TypeInfo, element_type_hash> element_types_map_t;
+        using map_t = std::unordered_map<element::Type_t, TypeInfo, element_type_hash>;
+        static const map_t map;
```

Comment:
Just a good-practices suggestion: variable names should not overlap with type names (`std::map` which becomes `map` if someone does `using namespace std`).

Reply:
Do you suggest I should add some protection for bad practice? :)
...
will do.

Reply:
In a way - yes, but using namespace std in .cpp files is quite common.